src/uct/base/uct_iface.h

```diff
@@ -551,6 +551,21 @@ typedef struct {
     }
 
 
+/**
+ * Helper macro to invoke the function from iface operations.
+ *
+ * @param _iface    UCT interface.
+ * @param _ops_type Type of iface operations.
+ * @param _func     Function to call.
+ * @param _ret      Return value.
+ */
+#define uct_iface_invoke_ops_func(_iface, _ops_type, _func, _args, _ret, ...) \
+    do { \
+        _ops_type *__ops = ucs_derived_of((_iface)->ops, _ops_type); \
+        _ret             = __ops->_func(__VA_ARGS__); \
+    } while (0)
```

Comment:
can we return _ret as the macro value
e.g ({   .... ; _ret; })

Reply:
done